document/RuleMetadataProfile.mdk

```diff
@@ -1 +1,165 @@
+
+A `ruleDescriptor` object may contain a property named `name` whose value is a string containing
+a readable rule identifier. A rule name is suitable for use as a key to other data or to
+otherwise identify a rule (e.g., within an in-source suppression) but is not guaranteed to
+be stable. In cases where the readable rule `name` contains implementation details that are
+invalidated over time, for example, a tool might alter a rule's readable name (while leaving
+the stable `id` property unchanged).
+
+~ IsoExample
+`"SpecifyMarshalingForPInvokeStringArguments"`
+~
+
+### `abstract` property { #ruleDescriptor-abstract }
+
+A `ruleDescriptor` object may contain a property named `abstract` whose value is a string containing
+a concise description of the rule. The `abstract` property should be a single sentence that is understandable
+when displayed in user interface contexts where the available space is limited to a single line of text.
+
+~ IsoExample
+`"Specify marshaling for P/Invoke string arguments"`
+~
+
+### `description` property { #ruleDescriptor-description}
+
+An `ruleDescriptor` object should contain a property named `description` whose value is a string that describes the rule.
+
+The `description` property should, as far as possible, provide details sufficient to enable resolution of issues. 
+
+The first sentence of the `description` property should provide a concise description of the rule,
+suitable for display in cases where UI is limited.
+Tools that construct `description` in this way need not provide a value for the `abstract` property.
+Tools that do not construct `abstract` in this way should provide a value for the `abstract` property,
+because otherwise, the initial portion of `description` that a viewer displays where UI space is limited
+might not be understandable.
+
+### `options` property { #ruleDescriptor-options }
+
+A `ruleDescriptor` object may contain a property named `options`, whose value is a JSON object
+consisting of an arbitrary set of string/value pairs.
+When present, the `options` objects shall describe the set of configurable options supported by the rule.
+
+The value within each string-value pair shall be a string, which may be the empty string.
+The value shall not be a JSON array or sub-object.
+
+### `formatSpecifiers` property { #ruleDescriptor-formatSpecifiers }
+
+A `ruleDescriptor` object may contain a property named `formatSpecifiers`, whose value is a JSON object
+consisting of an arbitrary set of string/value pairs.
+When present, the `formatSpecifiers` object shall describe a set of message formatting strings
+that have or may be returned by the tool.
+The string name of each format string shall be an identifier that uniquely identifies each member.
+The value within each string-value pair shall be a string that can be passed as a control parameter to
+string formatting API (_e.g._, `printf`) in order to construct a formatted message in combination with
```

Comment:
Yes, this is a real problem. The only way out here is to provide metadata that captures the convention for the format specifier (what flavor of printf, c# String.Format, etc.). This language/runtime affinity is not very desirable: should rules authors provide multiple flavors of formatted strings to support libraries that might attempt to hydrate them on different languages/platforms?

The alternate, dropping this idea entirely, is also not very desirable.